packages/mdx/lib/plugin/recma-jsx-rewrite.js

```diff
@@ -76,43 +76,43 @@ export function recmaJsxRewrite(options = {}) {
     walk(tree, {
       enter(_node) {
         const node = /** @type {Node} */ (_node)
+        const newScope = /** @type {Scope|undefined} */ (
+          // @ts-expect-error: periscopic doesn’t support JSX.
+          scopeInfo.map.get(node)
+        )
 
         if (
           node.type === 'FunctionDeclaration' ||
           node.type === 'FunctionExpression' ||
           node.type === 'ArrowFunctionExpression'
         ) {
-          fnStack.push({
+          const stackLength = fnStack.push({
             objects: [],
             components: [],
             tags: [],
             references: {},
             node
           })
-        }
 
-        let fnScope = fnStack[0]
+          // MDXContent only ever contains MDXLayout
+          if (
+            isNamedFunction(node, 'MDXContent') &&
+            newScope &&
+            !inScope(newScope, 'MDXLayout')
+          ) {
+            fnStack[stackLength - 1].components.push('MDXLayout')
```

Comment:
Any reason this is no longer `0`?
Even if in this case there’s only ever `0` on the stack, it’s probably better to be explicit about that and use `0`?

Reply:
Ping, just making sure if you saw this!